pkg/provider/provider.go

```diff
@@ -220,6 +220,11 @@ func buildTestEnvironment() { //nolint:funlen
 	operators := createOperators(data.Csvs, data.Subscriptions, data.AllInstallPlans, data.AllCatalogSources, false, false, true)
 	env.Operators = operators
 	logrus.Infof("Operators found: %d", len(env.Operators))
+	for _, p := range env.Pods {
+		if !(p.CreatedByDeploymentConfig()) {
```

Comment:
The idea of the check is the opposite: add a warning log trace when a pod has been deployed by a DeploymentConfig. Something like this:
```
for _, pod := range env.Pods {
   if pod.CreatedByDeploymentConfig() {
      logrus.Warnf("Pod %s has been deployed using a DeploymentConfig, please use Deployment or StatefulSet instead.", pod.String())
   }
}
```